### ISSUE TITLE
Trash assets failure with screenshots in nested folders

### ISSUE THREAD

**Issue by MikeMcC399** [cypress-io/cypress#29404]
If a test fails, then the option [trashAssetsBeforeRuns](https://docs.cypress.io/guides/references/configuration#Screenshots) with default `true` fails to delete previous screenshots on Windows 11 if there have been previous runs which produced screenshots in nested sub-folders of `cypress/screenshots`.

The error message is:

```text
DevTools listening on ws://127.0.0.1:59399/devtools/browser/d422c0e5-e2d8-456e-9962-e48e167e8351
Warning: We failed to trash the existing run results.

This error will not affect or change the exit code.

Error: Command failed: C:\Users\mikem\AppData\Local\Cypress\Cache\12.13.0\Cypress\resources\app\node_modules\trash\lib\win-trash.exe C:\Users\mikem\github\cypress-io\cypress-realworld-app\cypress\screenshots\ui

    at ChildProcess.exithandler (node:child_process:407:12)
    at ChildProcess.emit (node:events:527:28)
    at maybeClose (node:internal/child_process:1092:16)
    at Process.onexit (node:internal/child_process:302:5)
```

The script `dev` is opening the screenshots, which prevents Cypress `trash` from working.

## Steps to reproduce

Cypress: ~~12.13.0~~ 13.0.0
Operating system: Windows 11
Node.js: ~~16.16.0~~ 18.16.1

Create a new test spec `cypress/tests/demo/show-fail.spec.ts` with the following contents which will always fail when run:

```js
it("show test failing", function () {
  throw new Error("deliberate hard failure");
});
```

Execute

```bash
yarn
yarn dev
```
then in a separate terminal window, execute

```bash
yarn cypress:run
```

The new test should fail and generate 3 screenshots ~~and 1 video~~.

Kill the `dev`process (if necessary sign out and back in again) and repeat

```bash
yarn dev
```

and again in a separate terminal window, execute

```bash
yarn cypress:run
```

Note the error message (see above).

Attempt to delete `cypress/screenshots` in Windows Explorer.
Note message "Folder Access Denied / You'll need to provide administrator permission to delete this folder". Deletion is not possible.

Kill `yarn dev` process.

Attempt to delete `cypress/screenshots` in Windows Explorer.
This is successful.

## Suggestions

~~- When RWA is reworked, consider if this can be fixed.~~
- For Cypress itself, consider if the npm module [trash](https://www.npmjs.com/package/trash) is the best option for cleaning up temporary files (screenshots, etc.). [trash readme](https://github.com/sindresorhus/trash/blob/main/readme.md) says that the Linux implementation is not maintained. As shown by this issue it cannot delete open files. Perhaps the npm module [rimraf](https://www.npmjs.com/package/rimraf) would be a better choice for Cypress?

**Comment by MikeMcC399** [comment]
- I noticed this issue due to the flaky test https://github.com/cypress-io/cypress-realworld-app/issues/1332

**Comment by MikeMcC399** [comment]
Unfortunately it seems that this issue is unreliably reproducible 🙁 although it is still happening.

When it happens, screenshot copies build up, since they can't be deleted.

**Comment by MikeMcC399** [comment]
- Steps to reproduce are posted to https://github.com/cypress-io/cypress/issues/2876#issuecomment-2015322846. This adds a test to RWA which always fails.

**Comment by jennifer-shehane** [comment]
@MikeMcC399 It sounds like this issue should be moved to the main repo.

**Comment by MikeMcC399** [comment]
- There hasn't been any progress made in resolving this issue, which is still reproducible with Cypress 13.15.2, however, since the issue is fully covered by the open issue #2876, this issue can be closed as duplicate of #2876.
